app/views/recipes/show.html.erb

```diff
@@ -0,0 +1,21 @@
+<h1> <%=@recipe.name %></h1>
+<p>Complexity:</p><%= @recipe.complexity %>
+<p>Genre:</p><%= @recipe.genre %>
+<br>
+
+<h3>Ingredients:<h3>
+<% @ingredients.each do |ingredient|  %>
+<ul>
+<li><%= ingredient %></li><br>
+</ul>
+<%  end  %>
+
+<h3>Total Cost: <%= @cost %><h3>
+
+<h3>Add ingredient to recipe:</h3>
+<%# <%= form_with url: "/recipes/#{@recipe.id}", method: :post, local: true, data: {turbo: false} do |f|  %> 
+<%= form_with url: "/recipes/#{@recipe.id}", method: :post, local: true, data: {turbo: false} do |f|  %>
```

Comment:
To make the route for this form slightly more RESTful, consider `/recipes/:recipe_id/ingredients` to indicate that you're creating a join record between recipes and ingredients.

Reply:
ok so just to clarify, a route should only be performing one action? Or only route to a specific type of action?

Reply:
Routes can sometimes perform one action (i.e. a PUT and a PATCH can go to the same `update` action) but in this case, it's more RESTful to have a path that clearly indicates the 2 resources involved in the relationship you're setting up: ingredients and recipes. So then you path could be "/recipe_ingredients" or "/recipes/:id/ingredients"

Reply:
ahhh I see. I really need to stop seeing URLs as an identifier of a page where something is going but rather think of it as an association that can be name anything. Thanks for the clarification!